pkg/controller/signup_test.go

```diff
@@ -252,26 +247,41 @@ func (s *TestSignupSuite) TestInitVerificationHandler() {
 	}
 
 	s.Run("init verification success", func() {
-		assertInitVerificationSuccess("2268213044", "fd276563a8232d16620da8ec85d0575f", 1)
-	})
+		gock.New("https://api.twilio.com").
+			Persist().
+			Reply(http.StatusNoContent).
+			BodyString("")
+		defer gock.OffAll()
+		fakeClient, handler := prepareVerificationHandler(s.T(), userSignup)
 
-	s.Run("init verification success phone number with parenthesis and spaces", func() {
-		assertInitVerificationSuccess("(226) 821 3045", "9691252ac0ea2cb55295ac9b98df1c51", 2)
-	})
+		assertInitVerificationSuccess(handler, fakeClient, "2268213044", "fd276563a8232d16620da8ec85d0575f", 1)
 
-	s.Run("init verification success phone number with dashes", func() {
-		assertInitVerificationSuccess("226-821-3044", "fd276563a8232d16620da8ec85d0575f", 3)
-	})
-	s.Run("init verification success phone number with spaces", func() {
-		assertInitVerificationSuccess("2 2 6 8 2 1 3 0 4 7", "ce3e697125f35efb76357ed8e3b768b7", 4)
+		s.Run("init verification success phone number with parenthesis and spaces", func() {
+			assertInitVerificationSuccess(handler, fakeClient, "(226) 821 3045", "9691252ac0ea2cb55295ac9b98df1c51", 2)
+
+			s.Run("init verification success phone number with dashes", func() {
+				assertInitVerificationSuccess(handler, fakeClient, "226-821-3044", "fd276563a8232d16620da8ec85d0575f", 3)
+
+				s.Run("init verification success phone number with spaces", func() {
+					assertInitVerificationSuccess(handler, fakeClient, "2 2 6 8 2 1 3 0 4 7", "ce3e697125f35efb76357ed8e3b768b7", 4)
+				})
+			})
+		})
 	})
```

Comment:
I had to refactor the format because the sub-tests depend on the previous ones. Previously it wasn't a problem because the factory created a new instance of the service for every call of the `VerificationService` method.